sdk/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleMinMaxSumCount.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.metrics.aggregator;
+
+import com.google.errorprone.annotations.concurrent.GuardedBy;
+import io.opentelemetry.sdk.metrics.data.MetricData.Point;
+import io.opentelemetry.sdk.metrics.data.MetricData.SummaryPoint;
+import io.opentelemetry.sdk.metrics.data.MetricData.ValueAtPercentile;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.ThreadSafe;
+
+@ThreadSafe
+public final class DoubleMinMaxSumCount extends AbstractAggregator {
+
+  private static final AggregatorFactory AGGREGATOR_FACTORY =
+      new AggregatorFactory() {
+        @Override
+        public Aggregator getAggregator() {
+          return new DoubleMinMaxSumCount();
+        }
+      };
+
+  // The current value. This controls its own internal thread-safety via method access. Don't
+  // try to use its fields directly.
+  private final DoubleSummary current = new DoubleSummary();
+
+  public static AggregatorFactory getFactory() {
+    return AGGREGATOR_FACTORY;
+  }
+
+  @Override
+  void doMergeAndReset(Aggregator target) {
+    DoubleMinMaxSumCount other = (DoubleMinMaxSumCount) target;
+
+    DoubleSummary copy = current.copyAndReset();
+    other.current.update(copy);
```

Comment:
what about just:
`current.mergeAndReset(other.current)`

```java
    private void mergeAndReset(DoubleSummary target) {
      long count;
      double sum;
      double min;
      double max;
      lock.writeLock().lock();
      try {
        count = this.count;
        sum = this.sum;
        min = this.min;
        max = this.max;
        this.count = 0;
        this.sum = 0;
        this.min = 0;
        this.max = 0;
      } finally {
        lock.writeLock().unlock();
      }
      target.update(count, sum, min, max);
    }

    private void update(long count, double sum, double min, double max) {
      if (count == 0) {
        return;  // nothing to do
      }
      lock.writeLock().lock();
      try {
        this.count += count;
        this.sum += sum;
        this.min = Math.min(min, this.min);
        this.max = Math.max(max, this.max);
      } finally {
        lock.writeLock().unlock();
      }
    }
```

Reply:
I like this idea. I'll try it out.

Reply:
This is the only issue left